### PR TITLE
feat(frontend): Open debug runs from component

### PR DESCRIPTION
**Pull request by fatonramadani**
- Used a common store to communicate between components that are not in the same hierarchy

### CODE REVIEW

File: frontend/src/lib/components/apps/editor/AppEditorHeader.svelte [hunk 1]
```diff
@@ -57,13 +57,35 @@
...
+		const { data: componentId } = $eventBus[0]
+		jobsDrawerOpen = true
+
+		const job = $jobs.find((job) => job.component === componentId)
+		if (job) {
+			selectedJobId = job.job
+		}
+
+		eventBus.update((events) => events.slice(1))
+	}
```
Comment (rubenfiszel) [inline frontend/src/lib/components/apps/editor/AppEditorHeader.svelte]: An event bus seems quite complex compared to putting an:
`openDebugRun(componentId)`

any strong reasons for it ? What else do you see having there in the future ?

File: frontend/src/lib/components/apps/editor/ComponentHeader.svelte [hunk 2]
```diff
@@ -78,6 +85,7 @@
 					<Alert type="error" title="Error during execution">
 						<div class="flex flex-col">
 							<span> See "Debug Runs" on the top right for more details </span>
```
Comment (rubenfiszel) [inline frontend/src/lib/components/apps/editor/ComponentHeader.svelte]: I would remove this altogether and put the error directly, with the button underneath or above it's clear enough

File: frontend/src/lib/components/apps/editor/AppEditorHeader.svelte (hunk 1 above)
Comment (fatonramadani) [reply frontend/src/lib/components/apps/editor/AppEditorHeader.svelte]: It's a pattern that is already used in the app editor, for example opening the connect input menu.
I would only use one pattern for theses kind of cross-component communications

**Comment by rubenfiszel** [comment]
I would add a red border to runs with errors in the debug runs panel directly in this PR

File: frontend/src/lib/components/apps/editor/AppEditorHeader.svelte (hunk 1 above)
Comment (rubenfiszel) [reply frontend/src/lib/components/apps/editor/AppEditorHeader.svelte]: I still don't understand the benefit. This adds many layers of indirection and bug possibilities for no apparent benefits.

The alternative would be to literally just pass the function `openDebugRun(componentId)`  in a global context which is a pattern we use almost everywhere. That function can be passed up using component let bindings. It's a lot cleaner

**Comment by fatonramadani** [comment]
Reverted the eventBus and implemented it in an easier way:
https://user-images.githubusercontent.com/456655/214297339-4582cb86-f133-43af-9dea-2fd873caf83a.mov

File: frontend/src/lib/components/apps/editor/ComponentHeader.svelte (hunk 2 above)
Comment (fatonramadani) [reply frontend/src/lib/components/apps/editor/ComponentHeader.svelte]: Done

File: frontend/src/lib/components/apps/editor/AppEditorHeader.svelte (hunk 1 above)
Comment (fatonramadani) [reply frontend/src/lib/components/apps/editor/AppEditorHeader.svelte]: Done

**Pull request merged**